astropy/io/fits/hdu/hdulist.py

```diff
@@ -1043,7 +1062,6 @@ def writeto(
                     hdu._output_checksum = checksum
                     hdu._prewriteto()
                     hdu._writeto(hdulist._file)
-                    hdu._postwriteto()
```

Comment:
You cannot remove it like that, `_postwriteto` is called from various places and used in the compressed HDU. So instead for now I would just remove the content of `_postwriteto`.

Reply:
Tests fail if I remove both too...

Reply:
> or are you suggesting to keep the function definition without any action?

yes, this way I don't see any test fail.